immune_subtype_clustering/nf_stage.py

```diff
@@ -0,0 +1,66 @@
+from orca.services.nextflowtower import NextflowTowerOps
+from orca.services.nextflowtower.models import LaunchInfo
+
+
+def monitor_run(ops: NextflowTowerOps, run_id: str):
```

Comment:
Nit: Just for learning: take a look at `async` module like this: https://github.com/Sage-Bionetworks-Workflows/ntap-add5-scripts/pull/1.

That said, don't worry too much about reworking this, we can do it for the next workflow.  I also have a strong feeling `nf-stage` will be heavily re-used